### ISSUE TITLE
Attribute Respec issue?

### ISSUE THREAD

**Issue by Jayshua83** [epinter/tqrespec#6]
When Respeccing Attributes does the app look at the actual attribute points spent or at the total of each stat? I changed my game files to grant a bit more of each stat per point spent and I feel like TQRespec is giving me back more points than I should have.

**Comment by epinter** [comment]
The application only permits redistribution of points, never give more points you already have. There's no bug I'm aware of, but I will test again with more than one character and report to you.

**Comment by Jayshua83** [comment]
I was only asking since like I said I modified the amount of each stat I get per point spent and it seems like the app is giving more back then spent. I'll play around a bit more too.

**Comment by epinter** [comment]
Everything is working correctly. Removed points from STR and DEX from one char (left as available), entered game and added points back, the base attributes remained the same on both. With other character, removed points from Health and Int, entered game and added everything to Int, everything OK, then on the same character, removed from Int and added back to Health. Both characters are with same base attributes as initially. If you feel you have a problem, try to reproduce, I didn't see nothing wrong yet. The code for attributes redistribution are a bit old, from initial version (2018).

Don't forget that the points you need to look in game are the ones that appears in white color when you place mouse over the attribute, in blue are the bonus you received from skills, equipment, etc...

**Comment by Jayshua83** [comment]
This is still an issue. And it is related to how I modded my game. :(

The Respec tool is somehow looking at the Total Value of each Attribute to respec each one. As in It sees 1000 HP and it removes 40 HP and gives 1 Attribute point?

Instead of looking at a total number of attribute points spent in each attribute?

I modded my game to grant 50 Health/Energy instead of 40 per point. And each STR/INT/DEX point spent gives 5 stats instead of 4.

That is causing the tool to add more points than spent. lol

I just copied my character and respecced. She started with 240 point which is WAY more than I spent, but for testing I spent those 240 (in-game) 120 to Health and 120 to Energy. I left the game and respecced and ended up with 300 points. :| lol

**Comment by epinter** [comment]
Each stat point(available) is equal to 40 for health and 40 for energy, this value is hard coded, here: https://github.com/epinter/tqrespec/blob/ed4c825cc9cbddc8ee1428af5679bc8a0fd82097/src/main/java/br/com/pinter/tqrespec/util/Constants.java#L32
https://github.com/epinter/tqrespec/blob/ed4c825cc9cbddc8ee1428af5679bc8a0fd82097/src/main/java/br/com/pinter/tqrespec/util/Constants.java#L34

For attributes (STR,INT and DEX), 4 points, hardcoded here:
https://github.com/epinter/tqrespec/blob/ed4c825cc9cbddc8ee1428af5679bc8a0fd82097/src/main/java/br/com/pinter/tqrespec/util/Constants.java#L26
https://github.com/epinter/tqrespec/blob/ed4c825cc9cbddc8ee1428af5679bc8a0fd82097/src/main/java/br/com/pinter/tqrespec/util/Constants.java#L28
https://github.com/epinter/tqrespec/blob/ed4c825cc9cbddc8ee1428af5679bc8a0fd82097/src/main/java/br/com/pinter/tqrespec/util/Constants.java#L30

The **total value is calculated** by the game, and it is not saved, tools can't change the final value of each attribute.  What is saved is the base value, that you can see when you hover the mouse cursor over the attribute name.

**Comment by Jayshua83** [comment]
My Problem is I modded the BASE game like I have said.

So I'll just reset my mod file back to vanilla and do it another way so it doesn't screw up the respec tool when I use it.

Thanks.

**Comment by epinter** [comment]
Well, the tool was not designed to work with mods... What mod are you using ?

**Comment by Jayshua83** [comment]
My own mod. I made a mod that does A LOT of things.

Including what I said before... "I modded my game to grant 50 Health/Energy instead of 40 per point. And each STR/INT/DEX point spent gives 5 stats instead of 4."

So that ruins the tool unless it had an option built in to change the default values of each point. And that might be too much work to be worth it. I'll set my point values back to vanilla and give a little extra points to make up for it.
